apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -65,388 +61,47 @@
  *
  * @psalm-import-type Files_SharingShare from ResponseDefinitions
  */
-class ShareAPIController extends OCSController {
+class ShareAPIController extends ShareApiControllerFactory {
 
 	private ?Node $lockedNode = null;
-	private string $currentUser;
 
-	/**
-	 * Share20OCS constructor.
-	 */
 	public function __construct(
-		string $appName,
 		IRequest $request,
-		private IManager $shareManager,
-		private IGroupManager $groupManager,
-		private IUserManager $userManager,
-		private IRootFolder $rootFolder,
-		private IURLGenerator $urlGenerator,
-		private IL10N $l,
-		private IConfig $config,
-		private IAppManager $appManager,
-		private ContainerInterface $serverContainer,
-		private IUserStatusManager $userStatusManager,
-		private IPreview $previewManager,
-		private IDateTimeZone $dateTimeZone,
-		private LoggerInterface $logger,
-		private IProviderFactory $factory,
-		private IMailer $mailer,
-		?string $userId = null,
+		protected IManager $shareManager,
+		protected IGroupManager $groupManager,
+		protected IUserManager $userManager,
+		protected IRootFolder $rootFolder,
+		protected IURLGenerator $urlGenerator,
+		protected IL10N $l,
+		protected IConfig $config,
+		protected IAppManager $appManager,
+		protected ContainerInterface $serverContainer,
+		protected IUserStatusManager $userStatusManager,
+		protected IPreview $previewManager,
+		protected IDateTimeZone $dateTimeZone,
+		protected LoggerInterface $logger,
+		protected IProviderFactory $factory,
+		protected IMailer $mailer,
+		string $userId,
```

Comment:
```suggestion
		IManager $shareManager,
		IGroupManager $groupManager,
		IUserManager $userManager,
		IRootFolder $rootFolder,
		IURLGenerator $urlGenerator,
		IL10N $l,
		protected IConfig $config,
		IAppManager $appManager,
		ContainerInterface $serverContainer,
		IUserStatusManager $userStatusManager,
		IPreview $previewManager,
		IDateTimeZone $dateTimeZone,
		LoggerInterface $logger,
		protected IProviderFactory $factory,
		protected IMailer $mailer,
		?string $userId,
```

Ideally I would also reorder to keep promoted properties together, I would say at the end (so, start with parameters from parent constructor in the same order as parent, add you own after)